CorsixTH/Lua/entities/humanoids/staff/doctor.lua

```diff
@@ -57,39 +58,14 @@ function Doctor:tickDay()
 
   -- is self researcher in research room?
   if self:isResearching() then
-    self.hospital.research:addResearchPoints(1550 + 1000*self.profile.skill)
-  -- is self using lecture chair in a training room w/ a consultant?
+    self.hospital.research:addResearchPoints(1550 + 1000 * self.profile.skill)
+  -- Are we learning new skills today?
   elseif self:isLearning() then
-    -- Find values for how fast doctors learn the different professions from the level
-    local level_config = self.world.map.level_config
-    local surg_thres = 1
-    local psych_thres = 1
-    local res_thres = 1
-    if level_config and level_config.gbv.AbilityThreshold then
-      surg_thres = level_config.gbv.AbilityThreshold[0]
-      psych_thres = level_config.gbv.AbilityThreshold[1]
-      res_thres = level_config.gbv.AbilityThreshold[2]
-    end
-    local general_thres = 200 -- general skill factor
-
     local room = self:getRoom()
-    -- room_factor starts at 5 for a basic room w/ TrainingRate == 4
-    -- books add +1.5, skeles add +2.0, see TrainingRoom:calculateTrainingFactor
+    local consultant = room.staff_member
     local room_factor = room:getTrainingFactor()
-    -- number of staff includes consultant
-    local staff_count = room:getStaffCount() - 1
-    -- update general skill
-    self:trainSkill(room.staff_member, "skill", general_thres, room_factor, staff_count)
-    -- update special skill based on consultant skills
-    if room.staff_member.profile.is_surgeon >= 1.0 then
-      self:trainSkill(room.staff_member, "is_surgeon", surg_thres, room_factor, staff_count)
-    end
-    if room.staff_member.profile.is_psychiatrist >= 1.0 then
-      self:trainSkill(room.staff_member, "is_psychiatrist", psych_thres, room_factor, staff_count)
-    end
-    if room.staff_member.profile.is_researcher >= 1.0 then
-      self:trainSkill(room.staff_member, "is_researcher", res_thres, room_factor, staff_count)
-    end
+    local student_count = room:getStaffCount() - 1 -- less 1 for consultant in room
+    self:trainSkills(consultant, room_factor, student_count)
```

Comment:
Can you have more than one consultant in the room (eg one just graduated)?

Reply:
You can indeed. But `consultant` here comes from `room.staff_member` on line 65, where it's the one tied to the room. How it prevents clashing of consultants I'm not sure but they don't. But if you drop another consultant in the room, they take over.

Also, there's a bug where it seems `room.staff_member` may not change until the dismissed staff member leaves the room. But that's obviously not what we're addressing here.

Reply:
I just wondered if the `- 1` was always correct, since a consultant is not learning.

Reply:
Ah, that's another bug I found. For example, if the handyman walks in the room he also gets counted. At the moment, I've left it in under the idea perhaps the handymen is distracting the students with his presence :)

You can improve the count with various other checks, but not sure if it's worth it for the short inconvenience.

Reply:
You may want to explain this, so others are not tempted to change the `- 1`.